[docs] Update FAQ description explaining what Expo is used for (#19118)
expo/expo: docs/pages/introduction/faq.md

<Collapsible summary="What is Expo used for?">

> Expo is an [open-source framework](https://github.com/expo/expo/) for apps that run natively on Android, iOS, and the web. Expo brings together the best of mobile and the web and enables many important features for building and scaling an app like live updates, instantly sharing your app, and web support. The `expo` npm package is compatible with all React Native projects. Learn more about [what Expo offers](/introduction).

Reviewer: This makes expo look like it's an alternative to React Native which leads to things like “expo vs React Native”.

Author: ```suggestion
Expo is an [open-source framework](https://github.com/expo/expo/) for apps that run natively on Android, iOS, and the web. Expo brings together the best of mobile and the web and enables many important features for building and scaling an app like live updates, instantly sharing your app, and web support. The `expo` npm package also brings these features to any React Native project. Learn more about [what Expo offers](/introduction).
```
